### PR TITLE
Climb

### PR DESCRIPTION
**Pull request by DANIX147**

### CODE REVIEW

File: src/main/java/frc/robot/commands/UpAndDown.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        climb.rotateUpForward(direction);
+    }
+
+    @Override
+    public void end(boolean interrupted) {
+        climb.stop();
+    }
+
+    @Override
+    public boolean isFinished() {
```
Comment (Maayan-Luzon) [inline src/main/java/frc/robot/commands/UpAndDown.java:32]: use the limit switches to know when to stop the command.

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    public Climb() {
+        motor=new CANSparkMax(18, CANSparkLowLevel.MotorType.kBrushless);
+
+        forwardlimitswitch=  motor.getForwardLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+        reverselimitswitch= motor.getReverseLimitSwitch(SparkLimitSwitch.Type.kNormallyOpen);
+        motor.restoreFactoryDefaults();
+    }
+
+    public void rotateUpForward(boolean forwardOrReverse){
```
Comment (Maayan-Luzon) [inline src/main/java/frc/robot/subsystems/Climb.java:22]: change the name of this method. it is not clear what it does

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        motor.stopMotor();
+    }
+    public Boolean getf(){
+        return forwardlimitswitch.isPressed();
+    }
+
+    public boolean getr(){
+        return reverselimitswitch.isPressed();
+    }
+    public void print(){
```
Comment (Maayan-Luzon) [inline src/main/java/frc/robot/subsystems/Climb.java:36]: call the `print()` method in the periodic method.
`private void periodic(){}`

**Review by Maayan-Luzon — COMMENTED** [review]
good job. There are several thong to fix

File: src/main/java/frc/robot/commands/UpAndDown.java (hunk 1 above)
Comment (Maayan-Luzon) [reply src/main/java/frc/robot/commands/UpAndDown.java:32]: When moving up we want the forward limit switch to stop the movement and when moving down the reveres one need to stop it. 
For example, if I wish to raise the system, only when reaching the upper limit switch, the command should stop.

File: src/main/java/frc/robot/subsystems/Climb.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public void stop(){
+        motor.stopMotor();
+    }
+    public Boolean getf(){
+        return forwardlimitswitch.isPressed();
+    }
+
+    public boolean getr(){
+        return reverselimitswitch.isPressed();
+    }
```
Comment (AfraiCat) [inline src/main/java/frc/robot/subsystems/Climb.java:37]: Not really important but change the name `getf()` and `getr()` to `getForward()` and `getReverse()` so it's more understandable.

**Review by Maayan-Luzon — APPROVED** [review]
ok

**Review by Maayan-Luzon — APPROVED** [review]
after merging

**Pull request merged**